internal/features/stacks/events.go

```diff
@@ -234,6 +235,26 @@ func (f *StacksFeature) decodeStack(ctx context.Context, dir document.DirHandle,
 	}
 	ids = append(ids, eSchemaId)
 
+	validationOptions, err := lsctx.ValidationOptions(ctx)
+	if err != nil {
+		return ids, err
+	}
+	if validationOptions.EnableEnhancedValidation {
+		_, err = f.stateStore.JobStore.EnqueueJob(ctx, job.Job{
+			Dir: dir,
+			Func: func(ctx context.Context) error {
+				return jobs.SchemaModuleValidation(ctx, f.store, dir.Path())
+			},
+			Type:        operation.OpTypeSchemaModuleValidation.String(),
+			DependsOn:   ids,
```

Comment:
I think it's fine to have the job at this level currently, but I think we'll need to move it into defer of LoadStackMetadata soon, don't we? 🤔
(Mainly asking so that @dbanck can give some background 😄)

Reply:
Once we start adding inputs for components and they become available for validation, we need to depend on the `spawnedIds`. This ensures that the component metadata is fully loaded, before we run the validation.

And since this job depends on `PreloadEmbeddedSchema`, which also needs to move to `Defer`, this job needs to follow. _I think we moved it a few of days ago in another branch ?_

Reply:
Cool, thanks!

> I think we moved it a few of days ago in another branch ?

Yes, I did in https://github.com/hashicorp/terraform-ls/pull/1770

Reply:
Apologies, in my haste I habitually hit ready for review, but I really did intend on waiting for all of Ansgar's work to land before this was included so it could be properly ordered. Just waiting that